File: canteenn/client/ios/KeyboardObserver.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(KeyboardObserver, NSObject)

RCT_EXTERN_METHOD(isKeyboardShown:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: client/ios/KeyboardObserver.swift
import Foundation

@objc(KeyboardObserver)
class KeyboardObserver: NSObject {
  @objc
  func isKeyboardShown(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    DispatchQueue.main.async {
      if let keyWindow = UIApplication.shared.windows.first(where: { $0.isKeyWindow }),
         let _ = keyWindow.firstResponder {
        resolve(true)
      } else {
        resolve(false)
      }
    }
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}